plugins/pulp_rpm/plugins/importers/yum/upload.py

```diff
@@ -298,13 +297,15 @@ def _get_and_save_file_units(filename, processing_function, tag, conduit, repo_i
     :param conduit:  provides access to relevant Pulp functionality
     :type  conduit:  pulp.plugins.conduits.upload.UploadConduit
 
-    :param repo_id:  id of the repo into which unit will be uploaded
-    :type  repo_id:  str
+    :param repo: The repository to import the package into
+    :type  repo: pulp.server.db.model.Repository
     """
+    repo_id = repo.repo_id
     process_func = functools.partial(processing_function, repo_id)
     package_info_generator = packages.package_list_generator(filename, tag, process_func)
     for model in package_info_generator:
         model.save()
+        repo_controller.associate_single_unit(repo, model)
```

Comment:
I think it was introduced by this commit https://github.com/pulp/pulp_rpm/commit/17082bfc12abaff033485b57b3c0a591a015bcaa